File: leetcode/easy/palindrome-number.rb
def is_palindrome(x)
  rev = 0

  if x < 0 || (x % 10 == 0 and x != 0)
    return false
  end

  while x > rev
    digit = x % 10
    rev = rev * 10 + digit
    x = x / 10
  end

  if x == rev or (x == rev / 10)
    return true
  end

  return false
end

print(is_palindrome(1111))

File: leetcode/easy/palindrome-number.js
var isPalindrome = function(x) {
  let rev = 0;

  if (x < 0 || (x % 10 == 0 && x != 0)) {
    return false;
  }

  while(x > rev){
    let digit = x % 10;
    rev = rev * 10 + digit;
    x = Math.floor(x/10);
  }

  if (x == rev || (x == Math.floor(rev/10))) {
    return true;
  }

  return false;
};

console.log(isPalindrome(-121));
